### PR TITLE
refactor(swap): improve swap buy field performance

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-859](https://emurgo.atlassian.net/browse/YOMO-859)

[YOMO-859]: https://emurgo.atlassian.net/browse/YOMO-859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx [hunk 1]
```diff
@@ -223,14 +180,15 @@ const SelectableToken = ({tokenForList, wallet}: SelectableTokenProps) => {
...
 }: {
-  filteredTokensForList: Array<TokenForList>
-  allTokenInfos: Array<Balance.TokenInfo>
+  filteredTokensForList: Array<Balance.Token>
+  balances: Balance.Amounts
 }) => {
   const {search: assetSearchTerm, visible: isSearching} = useSearch()
+  const hasTokenBalance = Object.keys(balances).length === 1
 
-  if ((isSearching && assetSearchTerm.length > 0 && filteredTokensForList.length === 0) || allTokenInfos.length === 0)
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx:236]: When can`allTokenInfos.length === 0)` happen?

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx [hunk 2]
```diff
@@ -144,21 +93,22 @@ const TokenList = () => {
...
-        data={filteredTransformedList}
-        renderItem={({item: tokenForList}: {item: TokenForList}) => (
+        data={filteredTokenList}
+        renderItem={({item: token}: {item: Balance.Token}) => (
           <Boundary loading={{fallback: <AmountItemPlaceholder style={styles.item} />}}>
             <SelectableToken
-              tokenForList={tokenForList}
-              disabled={tokenForList.id !== wallet.primaryTokenInfo.id}
+              token={token}
+              disabled={token.info.id !== wallet.primaryTokenInfo.id}
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx]: This prop is defined as a type but not used in the component. I'm not sure if it's really necessary.

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx (hunk 1 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx:236]: Could it happen when the wallet has no tokens? But anyway, they should not be blocked from selecting the token to buy 🤔

Comment (banklesss) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx:236]: I have tested with a newly created wallet and the length value is 1 (ADA)

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx (hunk 2 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx]: This component was forked from the Send feature, some bits may be leftovers that don't work here. Like this disabled.

Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx]:
```suggestion
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx [hunk 3]
```diff
@@ -172,31 +121,38 @@ const TokenList = () => {
...
         closingText={strings.available}
       />
     </View>
   )
 }
 
-type SelectableTokenProps = {disabled?: boolean; tokenForList: TokenForList; wallet: YoroiWallet}
-const SelectableToken = ({tokenForList, wallet}: SelectableTokenProps) => {
+type SelectableTokenProps = {
+  disabled?: boolean
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx]:
```suggestion
```

**Review by stackchain — APPROVED** [review]

**Pull request merged**